Clarify user migration in 1.x to 2.0 automated upgrade process (#2344)
influxdata/docs-v2: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md

### User migration

`influxd upgrade` migrates existing 1.x users and their permissions **except** the following users:

- [1.x admin users](/{{< latest "influxdb" "v1" >}}/administration/authentication_and_authorization/#admin-users)
- [1.x non-admin users](/{{< latest "influxdb" "v1" >}}/administration/authentication_and_authorization/#non-admin-users)
  that have not been granted any privileges

{{< expand-wrapper >}}
{{% expand "Review 1.x user privileges" %}}
**To review 1.x users with admin privileges**, run the following against your InfluxDB 1.x instance:

```sql
SHOW USERS
```

> 

Reviewer: thoughts on adding: Any users labeled “admin” will not be migrated.

Author: 👍🏻

**To review the specific privileges granted to each 1.x user**, run the following for each user in your InfluxDB 1.x instance:

```sql
SHOW GRANTS FOR "<username>"
```
{{% /expand %}}

> {{< /expand-wrapper >}}

Reviewer: thoughts on adding: If no grants appear, the user labeled “admin” will not be migrated.

Author: 👍🏻

If using an admin user for visualization or Chronograf administrative functions,
**create a new read-only user before upgrading**:

##### Create a read-only 1.x user
```sh
> CREATE USER <username> WITH PASSWORD '<password>'
> GRANT READ ON <database> TO "<username>"
```

> InfluxDB 2.0 only grants admin privileges to the primary user created in the
> InfluxDB 2.0 setup process which runs at the end of the upgrade process.

Reviewer: ```suggestion
InfluxDB 2.0 only grants admin privileges to the primary user set up during the InfluxDB 2.0 upgrade.
```

> This provides you with the opportunity to reassess who to grant admin-level
> access to in your InfluxDB 2.0 setup.

Reviewer: ```suggestion
This provides you with the opportunity to reassess who to grant admin permissions when setting up InfluxDB 2.0.
```

Reviewer: Ideally, we'd include link to setting up InfluxDB, but given this doc isn't entirely task-based, it interferes with the workflow as is. For future thought...

### User migration

`influxd upgrade` migrates existing 1.x users and their permissions.

> However, it **does not** migrate the following users:

Reviewer: `influxd upgrade` migrates existing 1.x users and their permissions **except** the following users:

- administrative (admin) users

> - non-admin users without privileges

Reviewer: Users without admin permissions? 
Are admin users and non-admin users distinct in ways other than who has admin permissions? If yes, do we define this distinction?

Reviewer 2: The 1.x docs make a clear divide between admin and non-admin: https://docs.influxdata.com/influxdb/v1.8/administration/authentication_and_authorization/#user-types-and-privileges

My brain likes something like this better:
```suggestion
- administrative (admin) users
- non-admin users that have not been granted any privileges
```
So if people want to make sure all non-admins are migrated, they see there's an action they can take (granting privileges) to make it happen

To review users with admin permissions, run
[`SHOW USERS`](/{{< latest "influxdb" "v1" >}}/administration/authentication_and_authorization/#show-all-existing-users-and-their-admin-status)

> against your InfluxDB 1.x instance.

Reviewer: After finding users with admin permissions, manually migrate...?

Reviewer 2: I'm honestly not sure if there's a parallel 2.x entity that users could manually migrate to, other than the initial super-user token created by `influxd upgrade` or `influx setup`

> 

Reviewer 2: Should we also add a paragraph for "To review non-admin users with no permissions, ..."?

Author: Good call, added.
